### PR TITLE
feat(quest): include completions and points fields

### PR DESCRIPTION
**Pull request by iProdigy**
Same idea as #174 but for quest notifier

image

### CODE REVIEW

**Review by Felanbird — APPROVED** [review]
image

**Review by Felanbird — APPROVED** [review]
image

File: src/main/java/dinkplugin/notifiers/QuestNotifier.java [hunk 1]
```diff
@@ -5,15 +5,32 @@
...
+
+import javax.inject.Inject;
 
 import static net.runelite.api.widgets.WidgetID.QUEST_COMPLETED_GROUP_ID;
 
 public class QuestNotifier extends BaseNotifier {
 
+    @Varbit
+    @VisibleForTesting
+    static final int COMPLETED_ID = 6347, TOTAL_ID = 11877; // https://github.com/Joshua-F/cs2-scripts/blob/master/scripts/%5Bproc,questlist_completed%5D.cs2#L5
```
Comment (pajlada) [inline src/main/java/dinkplugin/notifiers/QuestNotifier.java:25]: nitpick: Could the comment be on as separate line? would make commit history slightly cleaner if we have to update one or the other

File: src/main/java/dinkplugin/notifiers/QuestNotifier.java [hunk 2]
```diff
@@ -5,15 +5,32 @@
...
+
+    @Varbit
+    @VisibleForTesting
+    static final int QP_TOTAL_ID = 1782; // https://github.com/Joshua-F/cs2-scripts/blob/master/scripts/%5Bproc,questlist_qp%5D.cs2#L5
```
Comment (pajlada) [inline src/main/java/dinkplugin/notifiers/QuestNotifier.java:29]: nitpick: Could the comment be on as separate line? would make commit history slightly cleaner if we have to update one or the other

File: src/main/java/dinkplugin/notifiers/QuestNotifier.java [hunk 3]
```diff
@@ -28,22 +45,40 @@ public void onWidgetLoaded(WidgetLoaded event) {
...
     }
 
     private void handleNotify(String questText) {
+        int completed = client.getVarbitValue(COMPLETED_ID);
+        int total = client.getVarbitValue(TOTAL_ID);
+        boolean validCount = completed > 0 && total > 0;
+
+        int points = client.getVarpValue(VarPlayer.QUEST_POINTS);
+        int possiblePoints = client.getVarbitValue(QP_TOTAL_ID);
+        boolean validPoints = points > 0 && possiblePoints > 0;
```
Comment (pajlada) [inline src/main/java/dinkplugin/notifiers/QuestNotifier.java:62]: nit: rename these local variables to match what they're called in QuestNotificationData

`completed` -> `completedQuests`
`total` -> `totalQuests`
and then `validCount` -> `validQuests`

`points` -> `questPoints`
`possiblePoints` -> `totalQuestPoints`

**Review by pajlada — CHANGES_REQUESTED** [review]
Small nitpicks, otherwise looks good

File: src/main/java/dinkplugin/DinkPlugin.java [hunk 4]
```diff
@@ -117,6 +117,7 @@ public void onUsernameChanged(UsernameChanged usernameChanged) {
 
     @Subscribe
     public void onGameStateChanged(GameStateChanged gameStateChanged) {
+        collectionNotifier.onGameState(gameStateChanged);
```
[the remark this answers is not in the document]
Comment (pajlada) [reply src/main/java/dinkplugin/DinkPlugin.java:120]: by "changes related to Collection Notifier" I specifically mean the onGameState thing, changing to use the new `formatProgress` in this PR is good :+1:

**Comment by iProdigy** [comment]
Is there any concern for people misinterpreting `%POINTS%`/`questPoints` as the incremental points earned from the given quest?

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,6 @@
 ## Unreleased
 
+- Minor: Include updated number of completions and points in quest notifications. (#178)
```
Comment (pajlada) [inline CHANGELOG.md:3]:
```suggestion
- Minor: Include updated number of quest completions and quest points in quest notifications. (#178)
```

**Review by pajlada — CHANGES_REQUESTED** [review]
You're right, `%POINTS%` could be misinterpreted. Best alternative I can come up with is `%CURRENT_POINTS%` but not sure

**Pull request merged**